integration-tests/contract-security-on=-chain-views.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { CONFIGS } from './config';
+
+/**   
+ * This "naive*" testcase assumes that the on-chain view stack is not correctly separated from the caller's stack. 
+ * Thus, this testcase assumes that there is only one stack, whereas the elements of the on-chain stack are on top. 
+ * Naive - meaning: WE just try it without thinking whether this test makes sense in regards with the used underlying architecture. 
+ * We think of the underlying architecture (type system, stack separation, etc.) as a black box.
+ * 
+ * TC-V-012: On-chain view - add instruction
+ * TC-V-013: On-chain view - dig instruction
+ * TC-V-014: On-chain view - dup instruction
+**/
+
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib;
+
+  describe(`Test contracts using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    it("Verify we can access the stack of the caller by using the instruction add.", async (done) => {
+      try {
+        const opGetter = await Tezos.contract.originate({
+          code: ` { parameter address;
+            storage nat;
+            code
+              {
+                UNPAIR;
+                UNIT;
+                VIEW "rogue" nat;
+                IF_NONE { FAIL } {};
+                ADD;
+                NIL operation;
+                PAIR;
+              };}`,
+          init: 'Unit'
```

Comment:
Not sure if it is expected: the storage should be a number (nat not unit)

Reply:
corrected. thanks!  now it is init: `0`